module/VuFind/src/VuFind/ILS/Driver/KohaRest.php

```diff
@@ -397,7 +397,14 @@ public function getStatuses($ids)
      */
     public function getHolding($id, array $patron = null, array $options = [])
     {
-        return $this->getItemStatusesForBiblio($id, $patron);
+        $biblio = $this->getBiblio($id);
+        $type   = $biblio['item_type'];
+        $options = [
```

Comment:
I'm a bit confused. Since we already support paging for holdings, why not add proper support for it here instead of overriding `$options`? Also, since module/VuFind/src/VuFind/ILS/Connection.php uses a setting with the same name, this is going to cause confusion later on as well.
So instead of handling the setting and changing `$options`, you could just check for `$options['itemLimit']` and  `$options['offset']` (see Alma::getHolding for example) and add the API parameters according to those. This would allow the user to page through the holdings instead of silently truncating the list.

If the itype-specific item limits are important, I'd add another setting like `itemLimitByType` and extend the `getConfig` method to check for it. If set, it would fetch the bib and copy any value found in itemLimitByType to the itemLimit setting to be returned from getConfig. Then itemLimit would still work as the default, and you'd only have to retrieve the bib if itemLimitByType is defined.

I realize that I might have explained myself really unclearly, so please ask if this just adds to the confusion. :)

Reply:
hi @EreMaijala thanks for your feedback.

I can certainly make those changes. one question i do have is because the koha-plugin-rest-di Biblio/Availability/Search->[_item_looper()](https://github.com/NatLibFi/koha-plugin-rest-di/blob/main/Koha/Plugin/Fi/KohaSuomi/DI/Koha/Biblio/Availability/Search.pm) does not accept the parameter of 'offset', I will need to add that to the koha-plugin-rest-di _item_looper()?

Thanks,
Alex

Reply:
thanks @EreMaijala , I've apart from my question about the Koha-plugin-rest-di plugin not accepting 'offset'. I think I have implemented all of your other points in my latest followup commit.

Could you please take a look?

Thank you very much.

Reply:
Argh, you're right. We also need the total count of items to support paging. But it's a bit hairy since these affect the biblio availability status too. I don't really like this all, but oh well. I'll see if I can add the offset quickly. We just need to add a note in KohaRest.ini that to use the feature you'll need version X of the DI plugin...

Reply:
Did you actually get the limit parameter to work either? As far as I can see it doesn't work in the DI plugin. The schema defines it as `limit_items`, but the code checks for `limit`...

Reply:
thanks for that @EreMaijala . 

Yes, I did get the limit parameter working. as the following screenshot shows I have 3 items returned for a journal biblio record due to VuFind making the following call to the DI plugin: 

![Screenshot from 2024-08-07 14-25-09](https://github.com/user-attachments/assets/7cb27826-85dd-4778-ac5b-c940ef64569d)

`2024-08-07T14:31:33+12:00 DEBUG (7): VuFind\ILS\Driver\KohaRest: [0.4933s] GET request <URL>/api/v1/contrib/kohasuomi/availability/biblios/1132644/search?limit=3`

Reply:
without that limit param then dozens more item records load.

Reply:
Well, that's odd. I'm only getting 400 Bad Request with "Malformed query string" because `limit` is not defined as a valid parameter for the endpoint. I've now got it working locally, but your success is puzzling!

Reply:
Maybe worth comparing versions? Was something added and then removed or broken subsequently?